packages/orchestration/src/examples/sendAnywhere.contract.js

```diff
@@ -0,0 +1,174 @@
+ * @import {TimerService, TimerBrand} from '@agoric/time';
+ * @import {LocalChain} from '@agoric/vats/src/localchain.js';
+ * @import {OrchestrationService} from '../service.js';
+ * @import {NameHub, Board} from '@agoric/vats';
+ * @import {Remote} from '@agoric/vow';
+ */
+
+/**
+ * @typedef {{
+ *   localchain: Remote<LocalChain>;
+ *   orchestrationService: Remote<OrchestrationService>;
+ *   storageNode: Remote<StorageNode>;
+ *   timerService: Remote<TimerService>;
+ *   agoricNames: Remote<NameHub>;
+ *   board: Remote<Board>;
+ * }} OrchestrationPowers
+ */
+
+const SingleAmountRecord = M.recordOf(M.string(), AmountShape, {
+  numPropertiesLimit: 1,
+});
+
+/**
+ * @param {ZCF} zcf
+ * @param {OrchestrationPowers & {
+ *   marshaller: Marshaller;
+ *   timerService: TimerService;
+ *   timerBrand: TimerBrand;
+ * }} privateArgs
+ * @param {Baggage} baggage
+ */
+export const start = async (zcf, privateArgs, baggage) => {
+  const zone = makeDurableZone(baggage);
+
+  const { makeRecorderKit } = prepareRecorderKitMakers(
+    baggage,
+    privateArgs.marshaller,
+  );
+
+  const chainHub = makeChainHub(privateArgs.agoricNames);
+  const makeLocalChainAccountKit = prepareLocalChainAccountKit(
+    zone,
+    makeRecorderKit,
+    zcf,
+    privateArgs.timerService,
+    privateArgs.timerBrand,
+    chainHub,
+  );
+
+  const { orchestrate } = makeOrchestrationFacade({
+    zcf,
+    zone,
+    chainHub,
+    // XXX prepare orchestration kinds separately
+    makeLocalChainAccountKit,
+    ...privateArgs,
+  });
+
+  let contractAccount;
+
+  const findBrandInVBank = async brand => {
+    const assets = await E(
+      E(privateArgs.agoricNames).lookup('vbankAsset'),
+    ).values();
+    const it = assets.find(a => a.brand === brand);
+    it || Fail`brand ${brand} not in agoricNames.vbankAsset`;
+    return it;
+  };
+
+  /** @type {OfferHandler} */
+  const sendIt = orchestrate(
+    'sendIt',
+    { zcf },
+    // eslint-disable-next-line no-shadow -- this `zcf` is enclosed in a membrane
+    async (orch, { zcf }, seat, offerArgs) => {
+      mustMatch(
+        offerArgs,
+        harden({ chainKey: M.scalar(), destAddr: M.string() }),
+      );
+      const { chainKey, destAddr } = offerArgs;
+      const { give } = seat.getProposal();
+      entries(give).length > 0 || Fail`empty give`;
+      const [[kw, amt]] = entries(give);
+      const { denom } = await findBrandInVBank(amt.brand);
+      const { [kw]: pmtP } = await withdrawFromSeat(zcf, seat, give);
+      const chain = await orch.getChain(chainKey);
+
+      // XXX ok to use a heap var crossing the membrane scope this way?
+      if (!contractAccount) {
+        const agoricChain = await orch.getChain('agoric');
+        contractAccount = await agoricChain.makeAccount();
+      }
+
+      const info = await chain.getChainInfo();
+      const { chainId } = info;
+      await E(contractAccount).deposit(await pmtP, amt);
```

Comment:
should theis be pipelined?

I really don't know what's allowed within an async flow

(and you point out, `contractAccount` is not remote itself)